src/ResourceManager/Storage/Commands.Management.Storage/StorageAccount/UpdateAzureStorageAccountNetworkRuleSet.cs

```diff
@@ -0,0 +1,177 @@
+// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Collections;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
+using Microsoft.Azure.Management.Storage;
+using Microsoft.Azure.Management.Storage.Models;
+using StorageModels = Microsoft.Azure.Management.Storage.Models;
+using Microsoft.Azure.Commands.Management.Storage.Models;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.Commands.Management.Storage
+{
+    [Cmdlet(VerbsData.Update, StorageAccountRuleSetNounStr, SupportsShouldProcess = true), OutputType(typeof(PSNetworkRuleSet))]
+    public class UpdateAzureStorageAccountRuleSetCommand : StorageAccountBaseCmdlet
+    {
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Resource Group Name.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Storage Account Name.")]
+        [Alias(StorageAccountNameAlias, AccountNameAlias)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Storage Account NetworkRule Bypass.")]
+        [ValidateNotNullOrEmpty]
+        public PSNetWorkRuleBypassEnum Bypass
+        {
+            get
+            {
+                return bypass == null ? PSNetWorkRuleBypassEnum.None : bypass.Value;
+            }
+            set
+            {
+                bypass = value;
+            }
+        }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Storage Account NetworkRule DefaultAction.")]
+        [ValidateNotNullOrEmpty]
+        public PSNetWorkRuleDefaultActionEnum DefaultAction
+        {
+            get
+            {
+                return defaultAction == null ? PSNetWorkRuleDefaultActionEnum.Allow : defaultAction.Value;
+            }
+            set
+            {
+                defaultAction = value;
+            }
+        }
+
+        [Parameter(
```

Comment:
In the future, I woudl consider a separate parameter set with string parameters for NetworkRuleId and IPAddressOrRange - this would make this cmdlet parallel with the create cmdlet